automol/util/vector.py

```diff
@@ -270,160 +257,155 @@ def _local_axes(xyz1=(0.0, 0.0, 0.0), xyz2=(0.0, 0.0, 1.0), xyz3=(0.0, 1.0, 0.0)
         cos = -1.0
     elif cos > 1.0:
-        assert numpy.allclose(cos, 1.0)
+        assert np.allclose(cos, 1.0)
         cos = 1.0
-    ang = numpy.arccos(cos)
+    ang = np.arccos(cos)
     return ang
 
 
-def dihedral_angle(xyz1, xyz2, xyz3, xyz4):
+def dihedral_angle(
+    xyz1: Sequence | np.ndarray,
+    xyz2: Sequence | np.ndarray,
+    xyz3: Sequence | np.ndarray,
+    xyz4: Sequence | np.ndarray,
+) -> float:
     """Measure the dihedral angle defined by four atoms.
 
     :param xyz1: 3D vector to point 1
-    :type xyz1: tuple, list, or numpy nd.array
     :param xyz2: 3D vector to point 2
-    :type xyz2: tuple, list, or numpy nd.array
     :param xyz3: 3D vector to point 3
-    :type xyz3: tuple, list, or numpy nd.array
     :param xyz4: 3D vector to point 4
-    :type xyz4: tuple, list, or numpy nd.array
-    :rtype: float
+    :return: Dihedral angle of four atoms
     """
-
     # Get the cosine of the angle
     uxyz21 = unit_direction(xyz2, xyz1)
     uxyz23 = unit_direction(xyz2, xyz3)
     uxyz32 = unit_direction(xyz3, xyz2)
     uxyz34 = unit_direction(xyz3, xyz4)
     uxyz123_perp = unit_perpendicular(uxyz21, uxyz23)
     uxyz234_perp = unit_perpendicular(uxyz32, uxyz34)
-    cos = numpy.dot(uxyz123_perp, uxyz234_perp)
+    cos = np.dot(uxyz123_perp, uxyz234_perp)
 
     # Get the sign of the angle
-    val = numpy.dot(uxyz123_perp, uxyz34)
+    val = np.dot(uxyz123_perp, uxyz34)
     val = max(min(val, 1.0), -1.0)
     sign = 2 * (val < 0) - 1
 
     # Before plugging it into the arccos function, make sure we haven't
     # Slightly run out of bounds
     if cos < -1.0:
-        assert numpy.allclose(cos, -1.0)
+        assert np.allclose(cos, -1.0)
         cos = -1.0
     elif cos > 1.0:
-        assert numpy.allclose(cos, 1.0)
+        assert np.allclose(cos, 1.0)
         cos = 1.0
 
-    dih = sign * numpy.arccos(cos)
-    dih = numpy.mod(dih, 2 * numpy.pi)
+    dih = sign * np.arccos(cos)
+    dih = np.mod(dih, 2 * np.pi)
     return dih
 
 
 # transformations
 def rotator(
-    axis: Vector, ang: float, degree: bool = False, orig_xyz: Optional[Vector] = None
+    axis: Vector, ang: float, degree: bool = False, orig_xyz: Vector | None = None
 ) -> Callable[[Vector], Vector]:
-    """Get a function for axis-angle rotations, optionally specifying the origin
+    """Get a function for axis-angle rotations, optionally specifying the origin.
 
     :param axis: Rotational axis (norm is ignored)
-    :type axis: Vector
     :param ang: Rotational angle
-    :type ang: float
     :param degree: _description_, defaults to False
-    :type degree: bool, optional
-    :param orig_xyz: _description_, defaults to None
-    :type orig_xyz: Optional[Vector], optional
     :return: _description_
-    :rtype: Callable[[Vector], Vector]
     """
-    orig_xyz = numpy.array([0.0, 0.0, 0.0] if orig_xyz is None else orig_xyz)
+    orig_xyz = np.array([0.0, 0.0, 0.0] if orig_xyz is None else orig_xyz)
```

Comment:
With the above change to the default value, this can be replaced with the following:
```
orig_xyz = numpy.array(orig_xyz)
```